src/ucp/core/ucp_worker.c

```diff
@@ -584,21 +584,27 @@ ucp_worker_iface_handle_uct_ep_failure(ucp_ep_h ucp_ep, ucp_lane_index_t lane,
     ucp_worker_h worker = ucp_ep->worker;
     ucp_wireup_ep_t *wireup_ep;
 
-    /* If the failure happened on AUX EP of CM lane on a server EP,
-     * it means that client closed its CM_WIREUP_EP/AUX_EP and it
-     * was detected before receiving WIREUP_MSG/ACK from a client or
-     * marking a server's EP as REMOTE_CONNECTED was scheduled on a
-     * progress, but not completed yet (CM_WIREUP_EP/AUX_EP is
-     * closed when moving an EP to REMOTE_CONNECTED state) */
+    /* If UCP EP isn't failed and the failure happened on AUX EP of CM lane:
+     * - on a server EP:
+     *   it means that client closed its CM_WIREUP_EP/AUX_EP and it was detected
+     *   before receiving WIREUP_MSG/ACK from a client or marking a server's EP
+     *   as REMOTE_CONNECTED was scheduled on a progress, but not completed yet
+     *   (CM_WIREUP_EP/AUX_EP is closed when moving an EP to REMOTE_CONNECTED
+     *   state)
+     * - on a client EP:
+     *   it means that server closed its CM_WIREUP_EP/AUX_EP and it was detected
+     *   when marking a client's EP as REMOTE_CONNECTED was scheduled on a
+     *   progress, but not completed yet (CM_WIREUP_EP/AUX_EP is closed when
+     *   moving an EP to REMOTE_CONNECTED state)
+     * So, no need to invoke the error handling flow for UCP EP to not fail
+     * normal connection establishment flow. If the real error was detected by
+     * AUX EP, other lanes will be failed too later. */
```

Comment:
does "real" mean that error can be "false negative"?  :) let's remove
maybe "if the error was detected by AUX EP, other lanes will be failed later during discarding"?

Reply:
> "if the error was detected by AUX EP, other lanes will be failed later during discarding"

no, it can be detected in ordinary flow, not in discarding

Reply:
> does "real" mean that error can be "false negative"? :) let's remove

i meant that `the real error` is an error due to peer failure or device/port down.

Reply:
> > "if the error was detected by AUX EP, other lanes will be failed later during discarding"
> 
> no, it can be detected in ordinary flow, not in discarding

but we do discart to force it anyway, right?

> > does "real" mean that error can be "false negative"? :) let's remove
> 
> i meant that `the real error` is an error due to peer failure or device/port down.

I think we should handle any type of failure, so the comment looks confusing

Reply:
> > > "if the error was detected by AUX EP, other lanes will be failed later during discarding"
> > 
> > 
> > no, it can be detected in ordinary flow, not in discarding
> 
> but we do discart to force it anyway, right?

yes, we WILL do discard after error detection on UCT EPs from UCP EP lanes

> I think we should handle any type of failure, so the comment looks confusing

we handle any type of failure, the only difference on which UCT EP it happens:
- AUX EP: don't invoke user's error handle callback and don't mark UCP EP as failed
- UCT EP from lanes: invoke user's error handle callback and mark UCP EP as failed

Reply:
@evgeny-leksikov do you still think that it should be rephrased?

Reply:
```
/* So, no need to invoke the error handling flow for CM_WIREUP_EP/AUX_EP, because the reason for this failure can be CM_WIREUP_EP/AUX_EP destroy on the remote side.  In case of transport-level error, we will detect it on transport lanes */
```

Reply:
👍 lets move it to begin of the comment before flow descriptions